Tufnel/retire from cm plumbing 1106 refactor

psparacino opened the pull request:
## Description

**This is a refactor of PR #1137** 

The retire link from the portfolio is not hooked up. To get to the retire page and test: `/portfolio/<token address here>/retire`

As `<Transaction />` is now being re-used, it now has an extra prop for the `spenderAddress`. I added it in the other components that needed it.

Given that the shared Modal is being used there are some style differences from the figma design. It might be worth adding some additional optional props to make the modal more customizable for different uses.

## Related Ticket

<!-- If your changes are related to a github ticket, please provide the issue number: -->

Resolves #1097 #1137

## Changes from Issue Requirements

**URL format**
<img width="807" alt="Screen Shot 2023-05-30 at 8 26 46 PM" src="https://github.com/KlimaDAO/klimadao/assets/72105234/4808f993-d3b2-4b49-bc22-2ef3e36aba15">



## Checklist

<!-- Check completed item: [X] -->

- [X] I have run `npm run build-all` without errors
- [X] I have formatted JS and TS files with `npm run format-all`

ladytrekker commented on carbonmark/components/Transaction/Approve.tsx:
```diff
@@ -67,23 +68,6 @@ export const Approve: FC<Props> = (props) => {
           }
           iconName={props.amount.token}
         />
-        {!!props.price && (
```
How come that the price is being removed here?
Can you point to the figma designs for that?

This Transaction Modal is used here:
- Create Listing
- Edit Listing
- Purchase a Listing

Editing a Listing always comes with the price.

ladytrekker replied:
| Before WITH PRICE | After WITHOUT PRICE  |
|---------|--------|
| <img width="400" alt="Bildschirmfoto 2023-05-19 um 14 37 53" src="https://github.com/KlimaDAO/klimadao/assets/95881624/35fa1eeb-c4c7-4179-a4d1-9e15482b8e7f">  | <img width="400" alt="Bildschirmfoto 2023-05-19 um 14 37 53" src="https://github.com/KlimaDAO/klimadao/assets/95881624/9fc65859-d8db-4766-ab17-7a9b9042e7a3">
 |

psparacino replied:
> Hey @psparacino
> 
> Having a quick glance at this PR as I am working on retirement from Project details (and may use your code here and there)
> 
> See some comments below.
> 
> Also I was not able to continue with the retirement on the preview link. You may be aware of that?
> 
> https://carbonmark-git-tufnel-retire-from-cm-plumbing-1-7961eb-klimadao.vercel.app/portfolio/0xd6ed6fae5b6535cae8d92f40f5ff653db807a4ea/retire <img alt="Bildschirmfoto 2023-05-19 um 14 49 55" width="1445" src="https://user-images.githubusercontent.com/95881624/239536634-4150055a-8a6e-41bd-92fa-0333db535b1d.png">

Thanks for taking a look; addressing comments individually below.

Re: retirement on the preview link, might be two reasons:

1) I was testing with a TC02 token; looks like you're using a C3T so I'll go back and confirm there are icons for all the carbon tokens (which looks like the reason for this failure).

2) When I tried using a TC02 token on the preview I still got a failure with `token` being undefined so maybe it has to do something with the env(?) in vercel? During development I had to specify the export `const DEFAULT_NETWORK = "mainnet";` otherwise it would default to testnet

psparacino replied:
Yes, price should not be removed on create/edit, so fixing that. it does look the design change is across the board though for all transactions; this is the create listing design in figma:
<img width="274" alt="create design" src="https://github.com/KlimaDAO/klimadao/assets/72105234/67f74b74-e1da-48b6-96e2-8bb90b1a8f3f">

Create Approve:
<img width="160" alt="create approve design" src="https://github.com/KlimaDAO/klimadao/assets/72105234/ab4958be-e783-4109-95c2-767584c05c6c">

ladytrekker commented on carbonmark/components/Transaction/index.tsx:
```diff
@@ -75,19 +80,12 @@ export const Transaction: FC<Props> = (props) => {
           status={props.status}
         />
       )}
-      {!!props.onGoBack && (
```
This Go Back Button was never included in any designs.
Still, wenn a transaction fails and the error message appears, the user should be able to go back to the former modal (not the transaction modal) to change the input values again.
This is the case for Editing a Listing for example whose inputs are also appearing in a modal (modal after modal). 

Just to be sure that this is still working.

psparacino replied:
This was the design I had with the 'Go Back' Button:
![Screen Shot 2023-05-19 at 10 19 32 AM](https://github.com/KlimaDAO/klimadao/assets/72105234/6564bb67-a068-4263-ac19-8eb32c168d87)

I had moved it up to the Retire Modal but I'll move it back to Transaction

ladytrekker commented on carbonmark/components/pages/Retire/RetireModal/styles.ts:
```diff
@@ -0,0 +1,181 @@
+`;
+
+export const details = css`
+  font-family: "DM Sans";
```
Just wondering about the font-style being addressed here directly.
We do have a typo-system for each variant of font.
For text there is the `Text` component which should be able to render the correct fonts.
Is this here not enough?

psparacino replied:
Will check that

ladytrekker reviewed (commented):
Hey @psparacino 

Having a quick glance at this PR as I am working on retirement from Project details (and may use your code here and there)

See some comments below.

Also I was not able to continue with the retirement on the preview link.
You may be aware of that?

https://carbonmark-git-tufnel-retire-from-cm-plumbing-1-7961eb-klimadao.vercel.app/portfolio/0xd6ed6fae5b6535cae8d92f40f5ff653db807a4ea/retire
<img width="1445" alt="Bildschirmfoto 2023-05-19 um 14 49 55" src="https://github.com/KlimaDAO/klimadao/assets/95881624/4150055a-8a6e-41bd-92fa-0333db535b1d">

ladytrekker commented on carbonmark/lib/listingsGetter.ts:
```diff
@@ -81,3 +82,15 @@ export const flagPrices = (prices: Price[]): PriceFlagged[] =>
     isPoolProject: true,
   }));
+
+export const createLinkWithLocaleSubPath = (
```
I would move this helper to /lib/createURLs

psparacino commented:
> Hey @psparacino
> 
> Having a quick glance at this PR as I am working on retirement from Project details (and may use your code here and there)
> 
> See some comments below.
> 
> Also I was not able to continue with the retirement on the preview link. You may be aware of that?
> 
> https://carbonmark-git-tufnel-retire-from-cm-plumbing-1-7961eb-klimadao.vercel.app/portfolio/0xd6ed6fae5b6535cae8d92f40f5ff653db807a4ea/retire <img alt="Bildschirmfoto 2023-05-19 um 14 49 55" width="1445" src="https://user-images.githubusercontent.com/95881624/239536634-4150055a-8a6e-41bd-92fa-0333db535b1d.png">

Thanks for taking a look; addressing comments individually below.

Re: retirement on the preview link, might be two reasons:

I was testing with a TC02 token; looks like you're using a C3T so I'll go back and confirm there are icons for all the carbon tokens (which looks like the reason for this failure).

When I tried using a TC02 token on the preview I still got a failure with token being undefined so maybe it has to do something with the env(?) in vercel? During development I had to specify the export const DEFAULT_NETWORK = "mainnet"; otherwise it would default to testnet

psparacino commented:
@ladytrekker Just summarizing my comments above:

Changes:
1) Moving Go Back Button back down to Transaction
2) optional prop in Transaction to show/hide price
3) Confirming all icons are accounted for
4) Font
5) Move link to helpers

I'll push these changes here.

The vercel failure might be a vercel settings issue, but will check again after these changes.

jabby09 commented:
@ShimonD-KlimaDAO this user flow is ready for initial QA

Note there was a related "plumbing" ticket (#1097) and the PR for all these changes is #1145 

PR link address to test from: https://carbonmark-git-tufnel-retire-from-cm-plumbing-1-7961eb-klimadao.vercel.app/

Other testing notes:
- Currently the RETIRE button link is not hooked up (this is part of the plumbing work in the PR)
- You can just input the url to test the retire flow, i.e. `/portfolio/<token address here>/retire`
- To get the token address you can right click copy the link destination on the pop up notification redirect modal and the token address will be in there

jabby09 reviewed (changes requested):
@psparacino (cc @ShimonD-KlimaDAO) initial testing feedback...

- [x] Default banner images are not using the darkened version so the white text is not readable. See a [project details](https://www.carbonmark.com/projects/VCS-1140-2015) page to see examples of the darkened default images. 
<img width="733" alt="Screenshot 2023-05-23 at 9 42 25 AM" src="https://github.com/KlimaDAO/klimadao/assets/103535195/1a10b4ef-35b0-4cd7-ac31-dcdafdd7c9d4">

- [x] This copy does not match design, please change to:
> Be careful not to include any sensitive personal information (such as an email address) in your retirement name or message. The information you enter here cannot be edited once it is submitted and will permanently exist on a public blockchain.
<img width="508" alt="Screenshot 2023-05-23 at 9 43 00 AM" src="https://github.com/KlimaDAO/klimadao/assets/103535195/7bd426e7-f9b6-4d6d-b6fc-807c9bcaac15">

- [x] Only the amount of tonnes is a required field so it needs the * label as per the design
![Screenshot 2023-05-23 at 12 08 07 PM](https://github.com/KlimaDAO/klimadao/assets/103535195/b5228577-51fa-4c0f-b05d-49f591d25341)

- [x] When tonnes are entered, the RETIRE CARBON button should be activated
<img width="401" alt="Screenshot 2023-05-23 at 9 44 05 AM" src="https://github.com/KlimaDAO/klimadao/assets/103535195/86b32557-30f0-480d-8dcc-bbc6fb39e601">

- [x] Beneficiary and retirement message are not required fields. Currently I had to enter these to get the RETIRE CARBON button to activate
<img width="389" alt="Screenshot 2023-05-23 at 9 44 49 AM" src="https://github.com/KlimaDAO/klimadao/assets/103535195/75ce4582-ca3c-48f2-8cdf-3d2bb5f2e016">

- [x] As you noted in discord, Confirm Retirement modal is not following the design implementation in other parts of the site, this needs to change to be consistent

- [x] Please change the copy on the APPROVE screen to:

> You are about to retire a carbon asset.
> 
> The first step is to grant the approval to transfer your carbon asset from your wallet to Carbonmark, the next step is to approve the actual transfer and complete your retirement.
> 
> Verify all information is correct and click 'approve' to continue.
<img width="561" alt="Screenshot 2023-05-23 at 11 53 29 AM" src="https://github.com/KlimaDAO/klimadao/assets/103535195/d0814a9c-dc04-4589-87aa-3a0f0b86c0f5">

- [x] Please change the copy of the SUBMIT screen to:

> The previous step granted the approval to transfer your carbon asset from your wallet to Carbonmark, your retirement has not been completed yet.
> 
> To finalize your retirement, verify all information is correct and then click 'submit' below.
<img width="550" alt="Screenshot 2023-05-23 at 11 54 40 AM" src="https://github.com/KlimaDAO/klimadao/assets/103535195/db8f4efe-5179-4ec8-8269-96da72488ab6">

- [x] Processing retirement spinner modal is not matching the design dimensions, it appears much longer width
<img width="648" alt="Screenshot 2023-05-23 at 9 47 06 AM" src="https://github.com/KlimaDAO/klimadao/assets/103535195/df925fd4-5566-4efb-b04a-1b7f99e89e70">

- [x] View transaction on polygonscan is not linking to the polygonscan transaction, it is linking to the retirement receipt
<img width="559" alt="Screenshot 2023-05-23 at 12 00 39 PM" src="https://github.com/KlimaDAO/klimadao/assets/103535195/5c5452b0-1a6f-410d-aafa-3383d6338035">

ShimonD-KlimaDAO commented:
1. According to #1097 , the link should be `/porfolio/retire/[project_token_0x]`, not `/porfolio/[project_token_0x]/retire`.

2. The approval tab on the transaction modal has this:
![image](https://github.com/KlimaDAO/klimadao/assets/99001851/fa219d20-3903-4c8f-a3d0-31a0f48bfae2)

3. The 'Submit' tab is explanation is not accurate. The approval does not grant access to move assets to the CM contracts (as with listings), but to the RA contracts.

4. Is "Balance (available to retire)" supposed to have the number of tonnes owned by the wallet? If not, I think this is a bit confusing at first glance. It makes it seem as if that's the number of tonnes the wallet owner can retire. 
![image](https://github.com/KlimaDAO/klimadao/assets/99001851/200e29a3-81bb-484c-b632-5a87975ed484)

5. Our capitalization of PolygonScan is inconsistent.

Version a (success modal in CM): 
![image](https://github.com/KlimaDAO/klimadao/assets/99001851/db627fe5-fc76-4030-b6c0-5eaff83442c7)

Version b (retirements page): 
![image](https://github.com/KlimaDAO/klimadao/assets/99001851/d633010c-fe61-43c0-b160-2e2f4e60ddbc)

Version c (Asset detail pane in this PR):
![image](https://github.com/KlimaDAO/klimadao/assets/99001851/b0c7ec70-f869-45eb-a302-f5a2b33a4141)

6. I [tried](https://carbonmark-git-tufnel-retire-from-cm-plumbing-1-7961eb-klimadao.vercel.app/portfolio/0x2f800db0fdb5223b3c3f354886d907a671414a7f/retire) putting in the BCT address to see what happens, and got a 404 error instead of bouncing me back to the Portfolio page as it does with tco2s and c3ts I don't own.

7. Some of the fields are off-center in mobile. There isn't a way to scroll right - it's just misaligned.
![image](https://github.com/KlimaDAO/klimadao/assets/99001851/12686f35-4b64-4746-85d7-2ac7a8211e96)

ladytrekker commented on carbonmark/components/Transaction/HighlightValue.tsx:
```diff
@@ -1,19 +1,23 @@
 }
 
-export const HighlightValue: FC<HighlightValueProps> = (props) => {
+export const HighlightValue: FC<RetireValueProps> = (props) => {
```
Why do you name it here RetireProps?

psparacino replied:
relic from a previous version, removed.

ladytrekker commented on carbonmark/components/Transaction/Submit.tsx:
```diff
@@ -33,60 +33,44 @@ export const Submit: FC<Props> = (props) => {
             </Text>
           }
-          value={props.amount.value}
+          value={props.amount.value || "0"}
```
In which case can `value` be undefined?

ladytrekker commented on carbonmark/components/Transaction/Submit.tsx:
```diff
@@ -33,60 +33,44 @@ export const Submit: FC<Props> = (props) => {
           }
           iconName={props.amount.token}
         />
-        {!!props.price && (
```
Here again => price is removed but still needed in other Transactions

ladytrekker commented on carbonmark/components/Transaction/index.tsx:
```diff
@@ -21,17 +20,25 @@ interface Props {
+  tokenIcon?: StaticImageData;
+  tokenName?: string;
+  onViewChange?: (newView: "approve" | "submit") => void;
+  showPrice?: boolean;
```
You can remove this prop as mentioned above.
Either a `price` is passed or not.

Also you are not using `showPrice` further down.

ladytrekker commented on carbonmark/components/Transaction/index.tsx:
```diff
@@ -21,17 +20,25 @@ interface Props {
     statusType === "userConfirmation" || statusType === "networkConfirmation";
 
+  useEffect(() => {
+    props.onViewChange?.(view);
```
I am wondering why we need an external `onViewChange` function?

The component has the prop `hasApproval` 
=> if this is `true` => show "Submit" view, else show "Approve"

If a Transaction has an approval should be passed down by the corresponding view (Edit Listing, Create Listing, Purchase a Listing, and here => Retire Modal)

ladytrekker reviewed (changes requested):
Hey @psparacino 

Thanks for the changes 🌟 
I was able to retire ✅ 

In the future, I would prefer that all the bigger changes which touch the Transaction Modal are happening in a separated PR (before or after this PR).
Now we need to ensure that all Transactions are still working.
Right now, Creating A Listing throws on this PR for example (see comments below).

I did not take a look at the implementation of the actual retirement transaction yet.
Just ping me again for a final review! Thanks 🙏

psparacino commented on carbonmark/components/CreateListing/index.tsx:
```diff
@@ -206,6 +207,7 @@ export const CreateListing: FC<Props> = (props) => {
           }}
           approvalText={<CreateApproval />}
           submitText={<CreateSubmit />}
+          spenderAddress={getAddress("carbonmark")}
```
looks like it was was same name, ethers import instead of the klimadao util. fixed.

ladytrekker commented on carbonmark/components/pages/Retire/RetireForm/index.tsx:
```diff
@@ -0,0 +1,460 @@
+                </label>
+
+                <div className="number_input_container">
+                  <input
```
There are already components for `Input` and `TextArea`. 
No need to write them again. 
See [here for example](https://github.com/KlimaDAO/klimadao/blob/staging/carbonmark/components/pages/Project/Purchase/PurchaseInputs.tsx#L59-L94)

Or what is the reason you did not use them?

ladytrekker commented on carbonmark/components/pages/Retire/RetireForm/styles.ts:
```diff
@@ -0,0 +1,293 @@
+  border: 2px solid #626266;
+  width: 100%;
+`;
+export const input = css`
```
As mentioned above, there are already components for inputs etc.

ladytrekker commented on carbonmark/pages/portfolio/[project_token_0x]/retire/index.tsx:
```diff
@@ -0,0 +1,64 @@
+        project,
+        fixedThemeName: "theme-light",
+      },
+      notFound: undefined,
```
Why `undefined`? Or rather remove it from here as you catch it further down?

ladytrekker commented on lib/constants/index.ts:
```diff
@@ -274,7 +274,9 @@ export const subgraphs = {
-  carbonmark: "https://api.thegraph.com/subgraphs/name/najada/marketplace-new", // TODO: ensure when switching drom testnet to mainnet that this is still correct!
+  carbonmark: "https://api.thegraph.com/subgraphs/name/najada/marketplace-new",
+  carbonmarkMatic:
+    "https://api.thegraph.com/subgraphs/name/najada/marketplace-matic", // TODO: ensure when switching drom testnet to mainnet that this is still correct!
```
Comment is in wrong place

ladytrekker reviewed (commented):
Hey @psparacino 

I checked the code again.
Please see my comments below 🙏

jabby09 commented:
@psparacino for this PR preview link, can we please have the RETIRE button hooked up to the correct path, it is a bit cumbersome testing to have to copy the token address from the Leaving Carbonmark modal.

<img width="688" alt="Screenshot 2023-05-26 at 9 27 44 AM" src="https://github.com/KlimaDAO/klimadao/assets/103535195/09ac70e3-2582-4331-a7c2-a6d8af4c3989">

jabby09 reviewed (changes requested):
@psparacino did another pass and identified more issues.

- [ ] Header image is not correct, the default image needs to be based on the Category. In this example, this is a Renewable Energy project but the image and Category in the retire form is "Other". Also the Verra (registry) pill is not rendering properly.

<img width="850" alt="Screenshot 2023-05-26 at 9 30 14 AM" src="https://github.com/KlimaDAO/klimadao/assets/103535195/5e793813-96d2-42ad-b610-cc50169fbf11">

- [ ] Header image not matching design. On the retire screen, the image is not the correct proportions. See the image below from the design, the height of the banner image pushes the bottom below the bottom of the Asset Details component. Also note the rounded corners at the top of the form/image.

![Screenshot 2023-05-26 at 10 24 14 AM](https://github.com/KlimaDAO/klimadao/assets/103535195/cd3f82ff-8885-470d-9b53-9692f103d654)

- [ ] The view transaction on PolygonScan link is not formatting correctly. It should be a link to the transaction, not the 0x address of the token. 

- For example, I did a retirement and this was the formatted link which results in a 404 error on polygonscan:
https://polygonscan.com/0x05e917686251e427034251087602da609e57f693/10/en

- The correct formatted link for that transaction I completed should be:
https://polygonscan.com/tx/0x3b7195d06acb7a082339e21b218718310da046ed3ef4b079c3a315d463b75948

<img width="755" alt="Screenshot 2023-05-26 at 9 35 48 AM" src="https://github.com/KlimaDAO/klimadao/assets/103535195/4186041c-7594-4e93-bde9-5708b117d638">

- [x] Please make adjustment to Asset Details design.

- Note the label change to `Available to retire` the value of which should be the tonnes available in this users wallet.
- Note that View on PolygonScan is moved to the bottom of the component.

![Screenshot 2023-05-26 at 12 01 48 PM](https://github.com/KlimaDAO/klimadao/assets/103535195/41c18c38-e795-427f-8b1f-aac57e8cbea4)

psparacino commented:
@jabby09 @ladytrekker 

Everything should be ready for a final review. 

@jabby09: To address your request about linking the retire button for convenience and Atmos's original request to not link so it can be merged without redirecting yet, I just made a temporary retire button next to the regular one. If all looks good on the review let me know and I'll take it out for the merge.

jabby09 commented:
@ShimonD-KlimaDAO not sure we have settled on the url structure (will confirm with @Atmosfearful) but this PR is updated and ready for another review / qa pass.

ladytrekker commented on carbonmark/components/pages/Retire/RetireModal/index.tsx:
```diff
@@ -0,0 +1,146 @@
+          onResetStatus={props.onResetStatus}
+          approvalText={<RetireApproval />}
+          submitText={<RetireSubmit />}
+          onGoBack={() => {
```
Here you do not need a Go Back button.
The user can simply close the modal and see the inputs again.

The Go Back button is needed when the input form was in the same modal before.
But this is not the case here.

Please add the Figma designs again to the PR description if I am missing something here.

psparacino replied:
this definitely makes sense but I had included the Go Back button as it was included in the design (below):

<img width="252" alt="Screen Shot 2023-05-30 at 10 55 15 AM" src="https://github.com/KlimaDAO/klimadao/assets/72105234/87203ee7-db9a-4660-a8e8-4009bc2536cc">

I agree there's not really a need for it however

ladytrekker commented on carbonmark/components/pages/Retire/RetireModal/styles.ts:
```diff
@@ -0,0 +1,196 @@
+import breakpoints from "@klimadao/lib/theme/breakpoints";
+import * as typography from "@klimadao/lib/theme/typography";
+
+export const container = css`
```
Not in use

ladytrekker commented on carbonmark/components/pages/Retire/RetireModal/styles.ts:
```diff
@@ -0,0 +1,196 @@
+  }
+`;
+
+export const viewSwitch = css`
```
Not in use

ladytrekker commented on carbonmark/components/pages/Retire/RetireModal/styles.ts:
```diff
@@ -0,0 +1,196 @@
+  padding: 0.4rem;
+`;
+
+export const switchButton = css`
```
Not in use

ladytrekker commented on carbonmark/components/pages/Retire/RetireModal/styles.ts:
```diff
@@ -0,0 +1,196 @@
+  }
+`;
+
+export const statusMessage = css`
```
Not in use

ladytrekker commented on carbonmark/components/pages/Retire/RetireModal/styles.ts:
```diff
@@ -0,0 +1,196 @@
+  }
+`;
+
+export const buttonRow = css`
```
Not in use

ladytrekker commented on carbonmark/components/pages/Retire/RetireModal/styles.ts:
```diff
@@ -0,0 +1,196 @@
+  display: flex;
+  justify-content: center;
+`;
+export const buttonColumn = css`
```
Not in use ..... and so on

ladytrekker reviewed (changes requested):
Thanks @psparacino for the update

I found a couple of clean-ups. Please see my comments below.

@Atmosfearful Please have a look too now and confirm the final URL structure again. See [comment](https://github.com/KlimaDAO/klimadao/pull/1145#discussion_r1210026935)

psparacino commented on carbonmark/components/pages/Portfolio/AssetProject/index.tsx:
```diff
@@ -72,6 +72,10 @@ export const AssetProject: FC<Props> = (props) => {
         />
         <CarbonmarkButton label={<Trans>Sell</Trans>} onClick={props.onSell} />
+        <CarbonmarkButton
+          href={`/portfolio/${props.asset.tokenAddress}/retire`}
```
Atmos had asked for this structure (different from the issue). It has changed back I can change it

ladytrekker replied:
Ok, let's wait for @Atmosfearful here

Atmosfearful replied:
https://discord.com/channels/841390338324824096/964698508751949824/1113557982203084800

psparacino commented on carbonmark/components/pages/Retire/RetireForm/index.tsx:
```diff
@@ -0,0 +1,441 @@
+      setIsLargeOrBelow(window.innerWidth <= breakpoints.large);
+    };
+
+    handleResize();
```
With the design I'm not sure how to handle the Asset Details sidebar responsiveness with only CSS? I'll take another look but I didn't see a way the first time through.

psparacino commented on carbonmark/components/pages/Retire/RetireModal/index.tsx:
```diff
@@ -0,0 +1,146 @@
+    if (statusType === "networkConfirmation" && txnView === "submit") {
+      setProcessingRetirement(true);
+    }
+  }, [props.status, txnView]);
```
Sounds good, I actually lifted processingRetirement up one component as well. It triggers the processing modal as soon as metamask opens instead of on submit, which I think makes sense. Catches the error in handleRetire to set it to false.

psparacino commented on carbonmark/components/pages/Retire/RetireForm/index.tsx:
```diff
@@ -0,0 +1,441 @@
+    window.innerWidth <= breakpoints.large
+  );
+
+  const [retirement, setRetirement] = useState({
```
Is it worth refactoring for continuity or should I just leave as is? Pretty simple form with only one validation.

ladytrekker replied:
That what just for your information.

ShimonD-KlimaDAO commented:
Looks great! A few remaining issues.

### From my original QA pass
(1) URL ordering - this is subjective, but my impression is that we are leaning towards `/retire/address`.
(3) This explanation in the Approve tab is inaccurate:
`The first step is to grant the approval to transfer your carbon asset from your wallet to Carbonmark, the next step is to approve the actual transfer and complete your retirement.`
Should be something like this:
`The first step is to grant the approval to transfer your carbon asset from your wallet to the Retirement Aggregator contract. The next step is to execute the retirement transaction.`

This explanation in the Submit tab is inaccurate:
`The previous step granted the approval to transfer your carbon asset from your wallet to Carbonmark, your retirement has not been completed yet.`
Should be something like this:
`The previous step granted the approval to transfer your carbon asset from your wallet to the Retirement Aggregator contract. Your retirement has not been completed yet.`

### New issue found
The portfolio page is off-center on mobile
![image](https://github.com/KlimaDAO/klimadao/assets/99001851/ad6135df-b087-446c-9c32-e53ca9a09875)

psparacino commented:
> Looks great! A few remaining issues.

**Great, thanks! Comments below.**
> 
> ### From my original QA pass
> (1) URL ordering - this is subjective, but my impression is that we are leaning towards `/retire/address`. 

**will double check with Atmos**

(3) This explanation in the Approve tab is inaccurate: `The first step is to grant the approval to transfer your carbon asset from your wallet to Carbonmark, the next step is to approve the actual transfer and complete your retirement.` Should be something like this: `The first step is to grant the approval to transfer your carbon asset from your wallet to the Retirement Aggregator contract. The next step is to execute the retirement transaction.`

**Will change**
> 
> This explanation in the Submit tab is inaccurate: `The previous step granted the approval to transfer your carbon asset from your wallet to Carbonmark, your retirement has not been completed yet.` Should be something like this: `The previous step granted the approval to transfer your carbon asset from your wallet to the Retirement Aggregator contract. Your retirement has not been completed yet.`

**Will change. Second sentence could maybe reference approving the txn in their wallet to complete retirement**

> 
> ### New issue found
> The portfolio page is off-center on mobile ![image](https://user-images.githubusercontent.com/99001851/242123775-ad6135df-b087-446c-9c32-e53ca9a09875.png)

**The Portfolio page itself I don't believe is actually part of this issue, but will see about fixing this and incorporating into this PR.**

jabby09 commented:
> > Looks great! A few remaining issues.
> 
> **Great, thanks! Comments below.**
> 
> > ### From my original QA pass
> > (1) URL ordering - this is subjective, but my impression is that we are leaning towards `/retire/address`.
> 
> **will double check with Atmos**
> 
> (3) This explanation in the Approve tab is inaccurate: `The first step is to grant the approval to transfer your carbon asset from your wallet to Carbonmark, the next step is to approve the actual transfer and complete your retirement.` Should be something like this: `The first step is to grant the approval to transfer your carbon asset from your wallet to the Retirement Aggregator contract. The next step is to execute the retirement transaction.`
> 
> **Will change**
> 
> > This explanation in the Submit tab is inaccurate: `The previous step granted the approval to transfer your carbon asset from your wallet to Carbonmark, your retirement has not been completed yet.` Should be something like this: `The previous step granted the approval to transfer your carbon asset from your wallet to the Retirement Aggregator contract. Your retirement has not been completed yet.`
> 
> **Will change. Second sentence could maybe reference approving the txn in their wallet to complete retirement**
> 
> > ### New issue found
> > The portfolio page is off-center on mobile ![image](https://user-images.githubusercontent.com/99001851/242123775-ad6135df-b087-446c-9c32-e53ca9a09875.png)
> 
> **The Portfolio page itself I don't believe is actually part of this issue, but will see about fixing this and incorporating into this PR.**

@psparacino @ShimonD-KlimaDAO do not change the wording / copy for approve and submit screens. The current copy was provided specifically by @sy-zygy so I'll need to check with him first. We can proceed with what we have, if it needs to change it can be a later enhancement.

ladytrekker reviewed (approved):
Thanks again @psparacino 🙇 

This PR needs to be rebased with staging again (see conflicting files)
Just rebase, accept the translation from staging and after that extract the translation on your branch again.

@Atmosfearful Please have a quick glance here too
and let's finish the decision on the URL structure.

Also keep in mind that this PR will change the styles of the Transaction modal for all transactions.

Atmosfearful commented on carbonmark/components/pages/Portfolio/AssetProject/index.tsx:
```diff
@@ -72,6 +72,10 @@ export const AssetProject: FC<Props> = (props) => {
           }}
         />
         <CarbonmarkButton label={<Trans>Sell</Trans>} onClick={props.onSell} />
+        <CarbonmarkButton
```
let's make sure to remove this before merging. Also, reminder when we do wire up the button, we need to pass next.js link so we can do a soft (client side) navigation, to avoid having to remount the entire app (which is causing the Login screen to reappear, which is related to this issue I've logged -> https://github.com/KlimaDAO/klimadao/issues/1174)

Atmosfearful reviewed (approved):
Looks good to me!
Just need to rebase.

Two follow up tickets:
https://github.com/KlimaDAO/klimadao/issues/1175
https://github.com/KlimaDAO/klimadao/issues/1174
